src/runner.js

```diff
@@ -84,13 +84,17 @@ assign(Runner.prototype, {
     const promise = Promise.using(this.ensureConnection(), function(connection) {
       hasConnection = true;
       runner.connection = connection;
-      const sql = runner.builder.toSQL()
-      const err = new Error('The stream may only be used with a single query statement.');
-      if (isArray(sql)) {
-        if (hasHandler) throw err;
-        stream.emit('error', err);
+      try {
```

Comment:
This change needs to be checked carefully...

Reply:
@elhigu The only difference I see now is that it will now emit an error even if _hasHandler_ is truthy. Would you consider that an unwelcome change?

Reply:
it was just a note for my futureself who has time to check this out better.